e2e_test/batch/transaction/read_only_multi_conn.slt

```diff
@@ -7,9 +7,15 @@ insert into t values (1), (2);
 statement ok
 flush;
 
+connection txn
+statement ok
+SET VISIBILITY_MODE TO checkpoint;
+
+connection txn
```

Comment:
Why do we need this? 👀

Reply:
After this PR, we won't provide consistency support for barrier read, and the CI with visibility_mode set to all will fail in this PR.

Reply:
Do you mean that read-only transaction does not guarantee any isolation when `visibility_mode` is `all`? If so, shall we ban user from doing this to avoid confusion?

Reply:
For simplicity, I change to send a notice to user when the user explicitly start read only transaction with visibility_mode is set to all.